### PR TITLE
fix: Close is called twice on PushScaler's deletion

### PR DESCRIPTION
**Pull request by WindNotStop**
Signed-off-by: ytz [1020560484@qq.com](mailto:1020560484@qq.com)

The PR is from(https://github.com/kedacore/keda/pull/3598#discussion_r955166185)

### CODE REVIEW

**Comment by JorTurFer** [comment]
Sorry, I'm checking where is closed the scaler before the line you have removed and I can't find it. Could you point me to it?

**Comment by WindNotStop** [comment]
> Sorry, I'm checking where is closed the scaler before the line you have removed and I can't find it. Could you point me to it?
The call chain is as follows:
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/pkg/scaling/cache/scalers_cache.go#L231
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/pkg/scaling/cache/scalers_cache.go#L227
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/pkg/scaling/scale_handler.go#L209
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/pkg/scaling/scale_handler.go#L113
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/controllers/keda/scaledobject_controller.go#L440
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/controllers/keda/scaledobject_finalizer.go#L37
https://github.com/kedacore/keda/blob/23bc75fc51a2aa43b96f830590ddf45dc0cef60c/controllers/keda/scaledobject_controller.go#L146

**Review by JorTurFer — APPROVED** [review]
You are right, thanks for the explanation 🙇 
LGTM!
WDYT @zroubalik ? Am I missing something?

**Comment by JorTurFer** [comment]
Ping @zroubalik

**Review by tomkerkhove — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 - **General:** Provide patch for CVE-2022-3172 vulnerability ([#3690](https://github.com/kedacore/keda/issues/3690))
 - **General:** Respect optional parameter inside envs for ScaledJobs ([#3568](https://github.com/kedacore/keda/issues/3568))
+- **General:** A problem with PushScaler ([#3881](https://github.com/kedacore/keda/issues/3881))
```
[the remark this answers is not in the document]
Comment (WindNotStop) [reply CHANGELOG.md]: Agree. What do you think it looks now?

**Review by zroubalik — APPROVED** [review]

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**